kstreams/test_utils/test_clients.py

```diff
@@ -168,3 +178,22 @@ async def getone(
             return consumer_record
 
         return None
+
+    def seek(self, *, partition: TopicPartition, offset: int) -> None:
+        # This method intends to have the same signature as aiokafka but with kwargs
+        # rather than positional arguments
+        if partition.topic in self.topics:
+            topic = TopicManager.get(name=partition.topic)
+
+            # if the topic is not consumed and the offset is <= that the
+            # current queue size we can try to seek
+            if not topic.consumed and offset <= topic.size():
+                for _ in range(0, offset):
+                    topic.get_nowait()
+                    topic.task_done()
+
+                    # it means that this consumer can consume
+                    # from the TopicPartition so we can add it
+                    # to the _assignment
+                    if partition not in self._assignment:
```

Comment:
do we like this? Technically, there shouldn't be a message on the partition, since we are saying the topic was defined to have 3 partitions, so sending a message to a partition that doesn't exist on the topic should have created an error.

Reply:
Yes. The documentation should say: *Each `Streams` in your application will have assign 3 partitions by DEFAULT`*. What can happens is that the `TestClient` can produce an event to any partition, it could be different to the defaults ones, let's say:

```python
 await client.send(topic_name, value=value, key=key, partition=10)
```

Then you could try to to a seek on the partition 10 (which is valid) and if the Stream was added on runtime and there are not more events for the partition 10, then it must be added to the `_assignment`, otherwise the method `stream.consumer.partitions_for_topic(topic_name)` will miss the partition, in this case `10`

Reply:
understood, but shouldn't the client.send on partition 10 throw an error?

If you create a kafka topic with 3 partitions, then try to send to partition 10, shouldn't it throw an error?

Reply:
in this scenario, I'm thinking about the workflow where the topic is created pre-send, rater than a create on send type scenario in kafka.